protocol/check.go

```diff
@@ -0,0 +1,33 @@
+
+var errInvalidSwapContract = errors.New("given contract address does not contain correct code")
+
+// CheckContractCode checks that the bytecode at the given address matches that of SwapFactory.sol.
+func CheckContractCode(ctx context.Context, ec *ethclient.Client, contractAddr ethcommon.Address) error {
+	code, err := ec.CodeAt(ctx, contractAddr, nil)
+	if err != nil {
+		return err
+	}
+
+	expectedCode := ethcommon.FromHex(expectedBytecode)
+
+	// sadly this is hard-coded for now since the forwarder address becomes part of the
+	// compiled bytecode.
+	if !bytes.Equal(expectedCode[:1473], code[:1473]) || !bytes.Equal(expectedCode[1543:], code[1543:]) {
+		return errInvalidSwapContract
```

Comment:
1543 - 1473 = 70 hex bytes. 35 binary bytes for the forwarder address?

Maybe compare `len(code)` with `len(expectedCode)` first so we know that those indices on `code` will never trigger a panic?

Reply:
the comment I added should explain what's in those bytes